source/RemoteWiring/RemoteDevice.cpp

```diff
@@ -92,37 +94,38 @@ RemoteDevice::analogRead(
 	Platform::String^ analog_pin_
 	)
 {
-	uint8_t parsed_pin = parsePinFromAnalogString(analog_pin_);
-	if (parsed_pin == static_cast<uint8_t>(-1))
-	{
-		return static_cast<uint16_t>(-1);
-	}
-	uint16_t val = -1;
-	uint8_t analog_pin_num = parsed_pin + _analog_offset;
+    uint16_t val = -1;
+	uint8_t parsed_pin = parsePinFromAnalogString( analog_pin_ );
 
-	{//critical section 
-		std::lock_guard<std::recursive_mutex> lock(_device_mutex);
+    {   //critical section 
+        std::lock_guard<std::recursive_mutex> lock( _device_mutex );
```

Comment:
OUt of curiosity, why the mutex? Won't the pin modes not change once you hit this point?